sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/sns/SnsIO.java

```diff
@@ -504,18 +504,18 @@ public void tearDown() {
       checkArgument(getCoder() != null, "withElementCoder() needs to called");
 
       return input
-          .apply(ParDo.of(new SnsAsyncWriterFn<>(this)))
+          .apply(ParDo.of(new SnsWriteAsyncrFn<>(this)))
           .setCoder(SnsResponseCoder.of(getCoder()));
     }
 
-    private static class SnsAsyncWriterFn<T> extends DoFn<T, SnsResponse<T>> {
+    private static class SnsWriteAsyncrFn<T> extends DoFn<T, SnsResponse<T>> {
 
-      private static final Logger LOG = LoggerFactory.getLogger(SnsAsyncWriterFn.class);
+      private static final Logger LOG = LoggerFactory.getLogger(SnsWriteAsyncrFn.class);
 
-      private final AsyncWrite<T> spec;
+      private final WriteAsync<T> spec;
       private transient SnsAsyncClient client;
 
-      SnsAsyncWriterFn(AsyncWrite<T> spec) {
+      SnsWriteAsyncrFn(WriteAsync<T> spec) {
```

Comment:
nit: `SnsWriteAsyncrFn` - I guess a letter "r" before "Fn" is redundant. Please remove it.